src/subprocess-win32.cc

```diff
@@ -107,7 +107,9 @@ bool Subprocess::Start(SubprocessSet* set, const string& command) {
 
   // Do not prepend 'cmd /c' on Windows, this breaks command
   // lines greater than 8,191 chars.
-  if (!CreateProcessA(NULL, (char*)command.c_str(), NULL, NULL,
+  std::wstring commands = Utf8ToWide(command).c_str();
+  LPWSTR command_line = const_cast <LPWSTR> (commands.c_str());
```

Comment:
`&commands[0]` would avoid the need for a const_cast

Reply:
Better would be to use C++11, and call the .data() function, but that's currently not allowed in Ninja.

> std::wstring commands = Utf8ToWide(command).c_str();

I have concerns about the lifetime of the return value of Utf8ToWide. Is this properly handled?

Can you please add some inline documentation to the code on how this is safe?

Reply:
You can use C++11 on Windows, I don't think we support older compilers there or will support them by the time this gets merged.

Talking about C++98 though, I don't think that it's guaranteed that wstring lays out its content continuously in memory, so `&commands[0]` would not be safe.

Btw: IIRC only the Unicode version of CreateProcess might actually modify the parameter, so we have to be careful now.

> I have concerns about the lifetime of the return value of Utf8ToWide. Is this properly handled?

The wstring ctor copies the data.

Reply:
What's the minimum supported compiler for Windows?

Is it the official Ninja project policy to support C++11 if it's inside of `#ifdef _WIN32` ?

Reply:
If it passes CI and no one complains, I would not test if code inside `#ifdef _WIN32` compiles with C++98 mode before merging a PR. Keep asking about official project policies and I will back down ;)

Reply:
> Keep asking about official project policies and I will back down ;)

Sorry, I'm not sure which of the following you mean:

1) If you keep asking for an official policy decision, I'll forbid C++11
2) If you keep asking for an official policy decision, I'll declare the project as C++11 fully.

I assume you mean 1, and not 2.

With that assumption, I don't think that's really fair. I just want to know one way or another what's allowed, so I can contribute PRs that won't be rejected out of hand.

If I can submit C++11 changes that are inside `#ifdef _WIN32` then I will do that. There are plenty of very-"low hanging fruit" C++11 changes that would improve Ninja's performance, and while I no longer use Ninja on Windows professionally, I'm still concerned about the entire codebase because I use it daily on non-Windows platforms. So any improvements to the code would be positive in my opinion.

Reply:
I can only speak for myself: I rarely review or merge Windows PRs as I don't use Ninja on Windows much. I make an exception for important changes like this one.